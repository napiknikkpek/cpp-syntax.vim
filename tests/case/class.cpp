class Foo {};
struct Foo {};
class Foo;
class Foo : Bar {};
class Foo : public Bar {};
class Foo : public virtual Bar {};
class Foo : Bar, A1, A2 {};
class Foo : public Bar, public A1, public A2 {};
class Foo : Bar::A1<decltype(foo{} + bar{})>, A1::A1, A2::A1 {};
class Foo : Bar<int, decltype(a > b)>, A1::A2<A3>, A2<3> {};
class {
} foo;
enum class Foo { A, B, C };
struct B final : A {
  void foo() override;
};
class ::ns::Base::X {};
class ns::Base<int>::X {};
class Base<T>::X {};
