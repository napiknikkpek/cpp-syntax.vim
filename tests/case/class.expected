aaaaa.ttt....
aaaaaa.ttt....
aaaaa.ttt.
aaaaa.ttt...ttt....
aaaaa.ttt...aaaaaa.ttt....
aaaaa.ttt...aaaaaa.aaaaaaa.ttt....
aaaaa.ttt...xxx..xx..tt....
aaaaa.ttt...aaaaaa.xxx..aaaaaa.xx..aaaaaa.tt....
aaaaa.ttt...ttt..xx..tt..xx..tt..tt....
aaaaa.ttt...ttt.zzz...tt..tt.tt...tt.c.....
aaaaa..
..xxx.
